internal/services/aadgraph/group_resource.go

```diff
@@ -201,22 +200,22 @@ func resourceGroupUpdate(d *schema.ResourceData, meta interface{}) error {
 		}
 
 		desiredMembers := *tf.ExpandStringSlicePtr(v.(*schema.Set).List())
-		membersForRemoval := slices.Difference(existingMembers, desiredMembers)
-		membersToAdd := slices.Difference(desiredMembers, existingMembers)
+		membersForRemoval := utils.Difference(existingMembers, desiredMembers)
+		membersToAdd := utils.Difference(desiredMembers, existingMembers)
 
 		for _, existingMember := range membersForRemoval {
 			var err error
 			var resp autorest.Response
-			log.Printf("[DEBUG] Removing member with id %q from Azure AD group with id %q", existingMember, d.Id())
+			log.Printf("[DEBUG] Removing member with id %q from Group with id %q", existingMember, d.Id())
 			attempts := 10
 			for i := 0; i <= attempts; i++ {
```

Comment:
Rather than hand rolling this, this we can use a StateFunc to achieve the same thing